composer/InstallMagoBinaryCommand.php

```diff
@@ -0,0 +1,268 @@
+    {
+        $version = InstalledVersions::getVersion(MagoPlugin::PACKAGE_NAME);
+
+        $response = $httpDownloader->get($this->buildGithubApiUri('/releases?per_page=99999999999999999'));
+        $json = $response->decodeJson();
+
+        foreach ($json as $release) {
+            if ($release['tag_name'] === $version) {
+                return $release['id'];
+            }
+        }
+
+        return 'latest';
+    }
+
+    /**
+     * @return array{tag: string, downloads: array<string, {file: string, url: string}>}
+     */
+    private function buildAssetsMapForRelease(HttpDownloader $httpDownloader, string $releaseId): array
+    {
+        $response = $httpDownloader->get($this->buildGithubApiUri('/releases/' . $releaseId));
+        $json = $response->decodeJson();
+
+        return [
+            'tag' => $json['tag_name'],
+            'downloads' => array_reduce(
+                $json['assets'] ?? [],
+                /**
+                 * @param array<string, string> $downloadMap
+                 * @param array{browser_download_url: string, name: string} $asset
+                 * @return array<string, string>
+                 */
+                static function (array $downloadMap, array $asset): array {
+                    if (!str_ends_with($asset['name'], '.tar.gz') && !str_ends_with($asset['name'], '.zip')) {
+                        return $downloadMap;
+                    }
+
+                    $platform = preg_replace('{^(.*)(\.tar\.gz|\.zip)$}', '$1', $asset['name']);
+
+                    $downloadMap[$platform] = [
+                        'file' => $asset['name'],
+                        'url' => $asset['browser_download_url'],
+                    ];
+
+                    return $downloadMap;
+                },
+                [],
+            ),
+        ];
+    }
+
+    private function buildGithubApiUri(string $path): string
+    {
+        return 'https://api.github.com/repos/carthage-software/mago' . $path;
+    }
+
+    /**
+     * @return array{platform: string, executable: string, storage_dir: string}
+     */
+    private function detectPlatformInfo(ProcessExecutor $process_executor, string $version): array
+    {
+        $arch_name = php_uname('m');
+        $arch = match ($arch_name) {
+            'x86_64', 'amd64' => 'x86_64',
+            'arm64', 'aarch64' => 'aarch64',
+            'armv7l' => 'armv7',
+            'i386', 'i486', 'i586', 'i686' => 'i686',
+            'ppc' => 'powerpc',
+            'ppc64' => 'powerpc64',
+            'ppc64le' => 'powerpc64le',
+            's390x' => 's390x',
+            default
+                => throw new \RuntimeException(
+                "Unsupported machine architecture: {$arch_name}. Please open an issue on GitHub.",
+            ),
+        };
+
+        $os = strtolower(php_uname('s'));
+        $vendor = 'unknown';
+        $os_suffix = '';
+        $executable_extension = '';
+
+        switch ($os) {
+            case 'windows':
```

Comment:
`php_uname('s')` will return `Windows NT` on windows so you should check for this string instead, though you should still make sure that the osname used in the filename is still `windows`

Reply:
Can you verify if I understood your comment correctly. 

Thanks for reviewing!

Reply:
Yes this looks fine to me :+1: I'll test again on windows tonight

Reply:
@Zuruuh is this good to go?

Reply:
Yeah I just confirmed it works now 😄